File: armstrong_num.py
def is_armstrong(n):
    temp = n
    digits = 0
    

    while temp > 0:
        digits += 1
        temp //= 10

    temp = n
    total = 0

    while temp > 0:
        digit = temp % 10
        total += digit ** digits
        temp //= 10

    return total == n

# Test
print(is_armstrong(153))   # True
print(is_armstrong(9474))  # True
print(is_armstrong(123))   # False

File: armstrong_num.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isArmstrong(int num) {
    int original = num;
    int digits = 0;

    // Count digits
    int temp = num;
    while (temp > 0) {
        digits++;
        temp /= 10;
    }

    // Calculate sum of digits^digits
    int sum = 0;
    temp = num;
    while (temp > 0) {
        int digit = temp % 10;

        // Instead of pow (floating-point), compute power manually
        int power = 1;
        for (int i = 0; i < digits; i++) {
            power *= digit;
        }

        sum += power;
        temp /= 10;
    }

    return sum == original;
}

int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;

    if (isArmstrong(num)) {
        cout << num << " is an Armstrong number." << endl;
    } else {
        cout << num << " is NOT an Armstrong number." << endl;
    }

    return 0;
}
